crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1491,19 +1489,39 @@ const fn is_ascii_identifier_start(c: char) -> bool {
 
 // Checks if the character c is a valid starting character as described
 // in https://docs.python.org/3/reference/lexical_analysis.html#identifiers
+#[cold]
```

Comment:
Can you try adding `#[inline(never)]` and to `is_unicode_identifier_continue` as well?